usd/src/usd_parser/USDTransforms.cc

```diff
@@ -0,0 +1,339 @@
+void GetTransform(
+  const pxr::UsdPrim &_prim,
+  const USDData &_usdData,
+  ignition::math::Pose3d &_pose,
+  ignition::math::Vector3d &_scale,
+  const std::string &_schemaToStop)
+{
+  std::vector<ignition::math::Pose3d> tfs;
+  GetAllTransforms(_prim, _usdData, tfs, _scale, _schemaToStop);
+  for (auto & rt : tfs)
+  {
+    _pose = rt * _pose;
+  }
+}
+
+//////////////////////////////////////////////////
+UDSTransforms ParseUSDTransform(const pxr::UsdPrim &_prim)
+{
+  auto variantGeom = pxr::UsdGeomGprim(_prim);
+  auto transforms = variantGeom.GetXformOpOrderAttr();
+
+  pxr::GfVec3f scale(1, 1, 1);
+  pxr::GfVec3f translate(0, 0, 0);
+  pxr::GfQuatf rotationQuad(1, 0, 0, 0);
+
+  UDSTransforms t;
+
+  pxr::VtTokenArray xformOpOrder;
+  transforms.Get(&xformOpOrder);
+  for (const auto &op : xformOpOrder)
+  {
+    if (op == kXFormOpScale)
+    {
+      auto attribute = _prim.GetAttribute(pxr::TfToken(kXFormOpScale));
+      if (attribute.GetTypeName().GetCPPTypeName() == kGfVec3fString)
+      {
+        attribute.Get(&scale);
+      }
+      else if (attribute.GetTypeName().GetCPPTypeName() == kGfVec3dString)
+      {
+        pxr::GfVec3d scaleTmp(1, 1, 1);
+        attribute.Get(&scaleTmp);
+        scale[0] = static_cast<float>(scaleTmp[0]);
+        scale[1] = static_cast<float>(scaleTmp[1]);
+        scale[2] = static_cast<float>(scaleTmp[2]);
+      }
+      t.SetScale(ignition::math::Vector3d(scale[0], scale[1], scale[2]));
+    }
+    else if (op == kXFormOpRotateZYX || op == kXFormOpRotateXYZ)
+    {
+      pxr::GfVec3f rotationEuler(0, 0, 0);
+      pxr::UsdAttribute attribute;
+      if (op == kXFormOpRotateZYX)
+      {
+        attribute = _prim.GetAttribute(pxr::TfToken(kXFormOpRotateZYX));
+      }
+      else
+      {
+        attribute = _prim.GetAttribute(pxr::TfToken(kXFormOpRotateXYZ));
+      }
+      if (attribute.GetTypeName().GetCPPTypeName() == kGfVec3fString)
+      {
+        attribute.Get(&rotationEuler);
+      }
+      else if (attribute.GetTypeName().GetCPPTypeName() == kGfVec3dString)
+      {
+        pxr::GfVec3d rotationEulerTmp(0, 0, 0);
+        attribute.Get(&rotationEulerTmp);
+        rotationEuler[0] = static_cast<float>(rotationEulerTmp[0]);
+        rotationEuler[1] = static_cast<float>(rotationEulerTmp[1]);
+        rotationEuler[2] = static_cast<float>(rotationEulerTmp[2]);
+      }
+      ignition::math::Quaterniond qX, qY, qZ;
+      ignition::math::Angle angleX(IGN_DTOR(rotationEuler[0]));
+      ignition::math::Angle angleY(IGN_DTOR(rotationEuler[1]));
+      ignition::math::Angle angleZ(IGN_DTOR(rotationEuler[2]));
+      qX = ignition::math::Quaterniond(angleX.Normalized().Radian(), 0, 0);
+      qY = ignition::math::Quaterniond(0, angleY.Normalized().Radian(), 0);
+      qZ = ignition::math::Quaterniond(0, 0, angleZ.Normalized().Radian());
+
+      if (op == kXFormOpRotateZYX)
+      {
+        std::swap(angleX, angleZ);
+      }
+      t.SetRotation((qX * qY) * qZ);
```

Comment:
Since ign-math already does Euler angles, my recommendation would be to use that instead of creating 3 quaternions here. I think this can be simplified as:
```
// SDFormat/ign-math uses extrinsic XYZ rotations, so it's the same as `rotateXYZ`
ignition::math::Quaterniond rot(IGN_DTOR(rotationEuler[0]), IGN_DTOR(rotationEuler[1]), IGN_DTOR(rotationEuler[2]))
  if (op == kxFormOpRotateZYX){
    rot.Invert();
  }
  t.SetRotation(rot);

```

See https://graphics.pixar.com/usd/release/api/class_usd_geom_xformable.html#details for USD rotation order

See http://sdformat.org/tutorials?tut=specify_pose&cat=specification& for SDFormat rotation order.